spectrum-offchain/src/box_resolver/rocksdb.rs

```diff
@@ -189,9 +189,11 @@ where
         <TEntity as OnChainEntity>::TStateId: 'a,
     {
         let db = self.db.clone();
+        let state_key = prefixed_key(STATE_PREFIX, &sid);
```

Comment:
This line and the one below was mistakenly removed in a prior commit.

Reply:
Not mistakenly, if we delete state prematurely, we will not be able to restore it in the case of rollback. See https://github.com/spectrum-finance/spectrum-offchain-ergo/blob/94c252effea03b3810ded751f7bc4c4f5fe7b95f/spectrum-offchain/src/event_sink/handlers/entity.rs#L53

Reply:
We will have to make cleanup mechanism later.

Reply:
I see. Not deleting state breaks the `test_rocksdb_invalidate` test, because `EntityRepo::get_last_[predicted|confirmed]` returns `None` iff state `state_key` no longer exists.

What do you think about using a new `INVALIDATED_STATE_KEY` prefix which is prepended to `state_key` when `invalidate` is called?

Btw, is it possible to reverse the effect of an `invalidate` call on a `state_key`?

Reply:
This issue was fixed in https://github.com/spectrum-finance/spectrum-offchain-ergo/pull/23